packages/genesys-spark-components/src/components/beta/gux-rich-text-editor/gux-rich-text-editor-action/gux-rich-text-editor-action-rich-style/gux-rich-text-editor-action-rich-style.tsx

```diff
@@ -0,0 +1,192 @@
+import { trackComponent } from '@utils/tracking/usage';
+import { buildI18nForComponent, GetI18nValue } from 'i18n';
+import translationResources from '../i18n/en.json';
+import { calculateDisabledState } from '../../gux-rich-text-editor.service';
+import { OnClickOutside } from '@utils/decorator/on-click-outside';
+import { whenEventIsFrom } from '@utils/dom/when-event-is-from';
+import { afterNextRender } from '@utils/dom/after-next-render';
+
+/** 
+@slot - for a collection of gux-rich-style-list-item elements.
+*/
+
+@Component({
+  tag: 'gux-rich-text-editor-action-rich-style',
+  styleUrl: 'gux-rich-text-editor-action-rich-style.scss',
+  shadow: { delegatesFocus: true }
+})
+export class GuxRichTextEditorActionRichStyle {
+  private i18n: GetI18nValue;
+  actionButton: HTMLElement;
+  listElement: HTMLGuxRichTextEditorListElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  @State()
+  expanded: boolean = false;
+
+  @Prop({ mutable: true })
+  disabled: boolean = false;
+
+  @OnClickOutside({ triggerEvents: 'mousedown' })
+  onClickOutside(): void {
+    this.expanded = false;
+  }
+
+  @Watch('disabled')
+  watchDisabled(disabled: boolean) {
+    if (disabled) {
+      this.expanded = false;
+    }
+  }
+
+  @Listen('keydown')
+  handleKeydown(event: KeyboardEvent): void {
+    const isButtonEvent = event.composedPath().includes(this.actionButton);
+    const isListEvent = event.composedPath().includes(this.listElement);
+
+    switch (event.key) {
+      case 'Escape':
+        if (isListEvent) {
+          event.preventDefault();
+          this.expanded = false;
+          this.actionButton.focus();
+        }
+        break;
+      case 'ArrowDown':
+      case 'Enter': {
+        if (isButtonEvent && !this.expanded) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusFirstListItem();
+        }
+        break;
+      }
+      case 'Tab':
+        this.expanded = false;
+        break;
+      case 'ArrowUp':
+        if (isButtonEvent && !this.expanded) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusLastListItem();
+        }
+    }
+  }
+
+  @Listen('keyup')
+  handleKeyup(event: KeyboardEvent): void {
+    switch (event.key) {
+      case ' ': {
+        if (event.composedPath().includes(this.actionButton)) {
+          event.preventDefault();
+          this.expanded = true;
+          this.focusFirstListItem();
+        }
+        break;
+      }
+    }
+  }
+
+  private focusFirstListItem(): void {
+    afterNextRender(() => {
+      void this.listElement.guxFocusFirstItem();
+    });
+  }
+
+  private focusLastListItem(): void {
+    afterNextRender(() => {
+      void this.listElement.guxFocusLastItem();
+    });
+  }
+
+  async componentWillLoad(): Promise<void> {
+    trackComponent(this.root);
+    this.i18n = await buildI18nForComponent(this.root, translationResources);
+    this.disabled = calculateDisabledState(this.root);
```

Comment:
I don't think this is the right thing to do. Toggling the parents disabled state will change this items disabled state permanently.
If this item is disabled and I toggle the parent from enabled to disabled to enabled this item should still be disabled.
If this does not make sense I can go on zoom to explain my issue better.

Reply:
That makes sense to me 👍  If it is set here it wont change due it only executing once. I think its probably better off attached to the class maybe. I have updated the change @daragh-king-genesys